### ISSUE TITLE
[CI] GetDataStreamsTransportActionTests testGetTimeSeriesMixedDataStream failing

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#96672]
**Build scan:**
https://gradle-enterprise.elastic.co/s/nmfsiknjr7dku/tests/:modules:data-streams:test/org.elasticsearch.datastreams.action.GetDataStreamsTransportActionTests/testGetTimeSeriesMixedDataStream

**Reproduction line:**
```
./gradlew ':modules:data-streams:test' --tests "org.elasticsearch.datastreams.action.GetDataStreamsTransportActionTests.testGetTimeSeriesMixedDataStream" -Dtests.seed=E4A62A9F4629F2D0 -Dtests.locale=tr-TR -Dtests.timezone=Etc/GMT-3 -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.datastreams.action.GetDataStreamsTransportActionTests&tests.test=testGetTimeSeriesMixedDataStream

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: iterable containing [(transformed to match "ds-1" and transformed to match iterable containing [".ds-ds-1-2023.06.06-000001", ".ds-ds-1-2023.06.06-000002", ".ds-ds-1-2023.06.06-000003"] and transformed to match iterable containing [<Tuple[v1=2023-06-07T16:34:23Z, v2=2023-06-07T20:34:23Z]>])]
     but: item 0: transformed to match iterable containing [".ds-ds-1-2023.06.06-000001", ".ds-ds-1-2023.06.06-000002", ".ds-ds-1-2023.06.06-000003"] transformed value item 0: was ".ds-ds-1-2023.06.07-000001"

  at __randomizedtesting.SeedInfo.seed([E4A62A9F4629F2D0:B1BE2FBA7E01F85B]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.datastreams.action.GetDataStreamsTransportActionTests.testGetTimeSeriesMixedDataStream(GetDataStreamsTransportActionTests.java:232)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

**Comment by martijnvg** [comment]
A test bug. Easy to fix and will do this later. I will mute it for now via #96673

**Comment by davidkyle** [comment]
Reopening: this test failed again on commit a8cf4d6006c912af97c1407ff0406bc4a9a9659c which contains the fix commit 055d4e88d4185cb6abd2bb3ad7f6e96f9e0df267

```
git rev-list a8cf4d6006c912af97c1407ff0406bc4a9a9659c | grep 055d4e88d4185cb6abd2bb3ad7f6e96f9e0df267
```

https://gradle-enterprise.elastic.co/s/exjujngqnlvam

```
Expected: iterable containing [(transformed to match "ds-1" and transformed to match iterable containing [".ds-ds-1-2023.06.12-000001", ".ds-ds-1-2023.06.12-000002", ".ds-ds-1-2023.06.12-000003"] and transformed to match iterable containing [<Tuple[v1=2023-06-11T23:16:36Z, v2=2023-06-12T03:16:36Z]>])] |  
-- | --
  | but: item 0: transformed to match iterable containing [".ds-ds-1-2023.06.12-000001", ".ds-ds-1-2023.06.12-000002", ".ds-ds-1-2023.06.12-000003"] transformed value item 2: was ".ds-ds-1-2023.06.11-000003"
```

Should the matcher be `containsInAnyOrder`?

**Comment by jdconrad** [comment]
I muted the test again for this in (https://github.com/elastic/elasticsearch/pull/96224). I couldn't get it to pass.
